test/mock_receiver.hpp

```diff
@@ -23,54 +23,59 @@
 
 namespace unifex_test {
 
```

Comment:
_Note:_ All of the changes in this file are to accommodate clang-cl not mangling `noexcept` into function types. This leads to linker errors. Rather than parameterizing template on `noexcept`-qualified function types, we pass `NoExcept` as a separate Boolean template parameter, forcing the compiler to generate different mangled names for `noexcept` and non-`noexcept` instantiations.